### PR TITLE
CAD-992, CAD-974, CAD-1130, CAD-1241, CAD-1231: txgen: shelley support + single shared pool + cleaner submission + updates

### PR DESCRIPTION
**Pull request by CodiePP**
Summary of the changes in the generator:

1. Implement global TPS limitation.
1. Implement submission from a shared Tx pool.
1. Reimplement the submission protocol in a simpler, more clear way.
1. Improve tracing.
1. End-of-run report, including per-thread submission TPS.
1. Make the generator actually exit at the end of submission.
1. More type safety.
1. Factor submission into local/node-to-node modules.
1. Support Shelley, except for the metadata-based extra Tx payload.
1. Update to the latest node.
1. Drop the explorer-API-based submission.

### CODE REVIEW

**Review by deepfire — APPROVED** [review]
LGTM!

**Comment by CodiePP** [comment]
bors cancel

**Review by deepfire — APPROVED** [review]
LGTM

File: benchmarks/cluster3nodes/run_tx_generator.sh [hunk 1]
```diff
@@ -28,6 +28,7 @@ run 'cardano-tx-generator' \
   --outputs-per-tx $outputstx \
   --tx-fee $txfee \
   --tps $tps \
+  --single-threaded \
```
Comment (CodiePP) [inline benchmarks/cluster3nodes/run_tx_generator.sh]: why should we use it in cluster3nodes?

File: benchmarks/cluster3nodes/run_tx_generator.sh [hunk 2]
```diff
@@ -28,7 +28,6 @@ run 'cardano-tx-generator' \
   --outputs-per-tx $outputstx \
   --tx-fee $txfee \
   --tps $tps \
-  --single-threaded \
```
Comment (CodiePP) [inline benchmarks/cluster3nodes/run_tx_generator.sh]: ah, now you remove "single-threaded" again

Comment (deepfire) [reply benchmarks/cluster3nodes/run_tx_generator.sh]: Right, for the period of time when we had the single-threaded submission, I thought it wise to recreate the cluster's conditions in the local testing environement.

**Review by MarcFontaine — APPROVED** [review]
LGTM

**Review by CodiePP — COMMENTED** [review]
LGTM

**Pull request merged**